processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -0,0 +1,210 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eks
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const (
+	// TypeStr is type of detector.
+	TypeStr = "eks"
+
+	k8sSvcURL = "https://kubernetes.default.svc"
+	/* #nosec */
+	k8sTokenPath      = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	k8sCertPath       = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+	authConfigmapPath = "/api/v1/namespaces/kube-system/configmaps/aws-auth"
+	cwConfigmapPath   = "/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info"
+	timeoutMillis     = 2000
+)
+
+// detectorUtils is used for testing the resourceDetector by abstracting functions that rely on external systems.
+type detectorUtils interface {
+	fileExists(filename string) bool
+	fetchString(httpMethod string, URL string) (string, error)
+}
+
+// This struct will implement the detectorUtils interface
+type eksDetectorUtils struct{}
+
+// This struct will help unmarshal clustername from JSON response
+type data struct {
+	ClusterName string `json:"cluster.name"`
+}
+
+var _ internal.Detector = (*Detector)(nil)
+
+// Detector for EKS
+type Detector struct {
+	utils detectorUtils
+}
+
+// Compile time assertion that eksDetectorUtils implements the detectorUtils interface.
+var _ detectorUtils = (*eksDetectorUtils)(nil)
+
+// NewDetector returns a resource detector that will detect AWS EKS resources.
+func NewDetector(_ component.ProcessorCreateParams, _ internal.DetectorConfig) (internal.Detector, error) {
+	return &Detector{utils: &eksDetectorUtils{}}, nil
+}
+
+// Detect returns a Resource describing the Amazon EKS environment being run in.
+func (detector *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
```

Comment:
This code looks familiar, maybe ported from Java? I think for the collector since we already use the k8s client library elsewhere, we can just use it here instead of writing all this fetch logic.

Reply:
Ah reread PR description and see it's copied from the Go SDK - but yeah in the collector I would probably expect this to just use the client library. Otherwise if the code can be deduped that sounds good too though.

Reply:
Can you clarify your plan here? I think it could be easier to just use the k8s client library to fix all of @tigrannajaryan's comments

Reply:
Yeah I'm assuming go-contrib didn't use k8s API to avoid dragging in all of k8s dependencies. I'm reluctant to add it since it will add more friction if we try to upstream it. However, I don't think it's completely unreasonable to say go-contrib and our use cases are different enough that they will just have separate implementations and each does the best for their particular case. 

Unless you have strong objections I say let's go with a minimally fixed up version in the hopes of consolidating with go-contrib.